src/webportal/src/app/job/job-view/fabric/task-attempt/task-attempt-list.jsx

```diff
@@ -0,0 +1,674 @@
+        res.push(
+          `On Windows:\nssh -p ${containerPorts.ssh} -i <your-private-key-file-path> root@${containerIp}\n`,
+        );
+        res.push(
+          `On Unix-like System:\nchmod 400 <your-private-key-file-path> && ssh -p ${containerPorts.ssh} -i <your-private-key-file-path> root@${containerIp}\n\n`,
+        );
+        res.push(
+          `If you are using a different username in your docker, please change "root" to your pre-defined username.`,
+        );
+      } else {
+        res.push('This job does not contain SSH info.');
+        res.push(
+          'If you want to use SSH, please enable it in the "Tools -> SSH" Section on the Job Submission Page.',
+        );
+      }
+      this.setState({
+        monacoProps: {
+          value: res.join('\n'),
+          options: {
+            wordWrap: 'off',
+            readOnly: true,
+          },
+        },
+        monacoTitle: `SSH to ${id}`,
+      });
+    }
+  }
+
+  getTaskPropertyFromColumnKey(item, key) {
+    if (key === 'exitType') {
+      return !isNil(item.containerExitSpec) &&
+        !isNil(item.containerExitSpec.type)
+        ? item.containerExitSpec.type
+        : null;
+    }
+    return item[key];
+  }
+
+  onRenderRow(props) {
+    return (
+      <DetailsRow
+        {...props}
+        styles={{
+          root: {
+            color: theme.palette.black,
+          },
+        }}
+      />
+    );
+  }
+
+  render() {
+    const { monacoTitle, monacoProps, monacoFooterButton, items } = this.state;
+    return (
+      <div>
+        <DetailsList
+          styles={{ root: { overflow: 'auto' } }}
+          columns={this.getColumns()}
+          disableSelectionZone
+          items={items}
+          layoutMode={DetailsListLayoutMode.justified}
+          selectionMode={SelectionMode.none}
+          onRenderRow={this.onRenderRow}
+        />
+        {/* Monaco Editor Panel */}
+        <MonacoPanel
+          isOpen={!isNil(monacoProps)}
+          onDismiss={this.onDismiss}
+          title={monacoTitle}
+          monacoProps={monacoProps}
+          footer={monacoFooterButton}
+        />
+      </div>
+    );
+  }
+
+  getColumns() {
+    const defaultColumns = [
+      {
+        key: 'taskAttemptIndex',
+        name: 'Task Attempt Index',
+        minWidth: 120,
+        headerClassName: FontClassNames.medium,
+        isResizable: true,
+        onRender: (item, idx) => {
+          return (
+            <div className={FontClassNames.mediumPlus}>{item.attemptId}</div>
+          );
+        },
+      },
+      {
+        key: 'taskAttemtState',
+        name: 'Task Attempt State',
+        minWidth: 120,
+        headerClassName: FontClassNames.medium,
+        isResizable: true,
+        onRender: item => (
+          <StatusBadge status={capitalize(item.attemptState)} />
+        ),
+      },
+      {
+        key: 'taskAttemptExitCode',
+        name: 'Task Attempt Exit Code',
+        minWidth: 150,
```

Comment:
Seems Task Attempt Exit Code in task page and Task Exit code in job page cannot be sort by col?

Reply:
Yes, this feature is not included in new design for time reason.